File: pset2/caesar/caesar.c
/*
Implement a program that encrypts messages using Caesar’s cipher, per the below.

$ ./caesar 13
plaintext:  HELLO
ciphertext: URYYB
*/

//\[c_i = (p_i + k) \bmod 26\]
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    string key[10];
    int n = strlen(argv[1]);
    if (argc == 2 && (isalpha(argv[1])))
    {
        for (int i = 0; i<n;i++)
        {
            char c = ((argv[1][i]) - 65);
            key[i] = &c;
        }

        printf("plaintext: ");
        string plaintext = get_string();

        printf("ciphertext: ");
        
        //iterate over the several key values
        for(int k = 0;k<n;k++)
        {
            int kvalue = *key[k];
            
            // iterate over strings in argv
            for (int j = 0,m = strlen(plaintext); j < m; j++)
            {
                int c = plaintext[j];
                if (isupper(c)) // if the char is capitalized
                {
                    c-= 65; // to alpha
                    c = ((c+kvalue)%26); //encription
                    c+=(65); //back to ascii
                    printf("%c", c);
                }
                else if(islower(c))
                {   c-= 97;
                    c = ((c+kvalue)%26); // encription
                    c+= 97;
                    printf("%c", c);
                }
                else
                {
                    printf("%c", c);
                    k--; // so it doesnt count
                }
            }
        }
        
        printf("\n");
    }
    else
    {
        printf("Error, please submit a non-negative integer single command-line argument\n");
        return 1;
    }
}

File: pset6/caesar.py
# Implement a program that encrypts messages using Caesar’s cipher, per the below.

# $ ./caesar 13
# plaintext:  HELLO
# ciphertext: URYYB

import cs50
import sys


def main():
    if len(sys.argv) == 2: #argc
        #get the key
        key = int(sys.argv[1])
        #get plaintext
        print("plaintext: ", end ="")
        ptext = cs50.get_string()
        #ENCIPHER
        #for each character in plaintext string
        for c in ptext:
            #if alphabetical shift character acc with key, preserving case
            if c.isalpha():
                if c.isupper:
                    print(chr(((ord(c) - 65 + key) % 26) + 65), end ="")
                if c.islower():
                    print(chr(((ord(c) - 65 + key) % 26) + 65), end = "")
    else:
        print("Usage: python caesar.py [KEY]")
    print()

if __name__ == "__main__":
    main()
    
